### PR TITLE
feat/1047 change how model props ymls are created when running generate properties

### PR DESCRIPTION
**Pull request by BAntonellini**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +135,24 @@ how the resources are generated.
 --update-strategy
 # Action to perform when a property file already exists: 'update', 'recreate', 'fail', 'ask' (per file)
 ```
+
+`dbt-coves generate properties` supports the following args:
+
+```console
```
Comment (ssassi) [inline README.md:141]: You should use `shell` https://github.com/github/linguist/blob/master/lib/linguist/languages.yml

File: README.md [hunk 2]
```diff
@@ -135,6 +135,24 @@ how the resources are generated.
...
+# Where models yml files will be generated, default: 'models/staging/{{schema}}/{{relation}}.yml'
+```
+
+```console
+--update-strategy
+# Action to perform when a property file already exists: 'update', 'recreate', 'fail', 'ask' (per file)
+```
+
+```console
+--model
```
Comment (ssassi) [inline README.md:152]: if it could process more than one model, it should be call `models`

File: dbt_coves/tasks/generate/base.py [hunk 3]
```diff
@@ -106,20 +109,27 @@ def get_metadata(self):
...
 
-    def render_templates(
-        self, relation, columns, destination, options=None, json_cols=None
-    ):
+    def get_config_value(self, key):
+        return self.coves_config.integrated["generate"][self.args.task][key]
+
+    def generate_template(self, destination_path, relation):
+        template_context = dict()
+        if "{{schema}}" in destination_path.replace(" ", ""):
```
Comment (ssassi) [inline dbt_coves/tasks/generate/base.py:125]: Replacing " " by "" in a path is a bug. In windows specially, it's very common to have folder names with spaces.

Comment (ssassi) [reply dbt_coves/tasks/generate/base.py:125]: You should use jinja.render

File: README.md (hunk 1 above)
Comment (BAntonellini) [reply README.md:141]: So all current 'console' command usages must be 'shell', right?

File: dbt_coves/tasks/generate/base.py [hunk 4]
```diff
@@ -156,3 +166,194 @@ def get_metadata_columns(self, relation, cols):
...
+        template=None,
+    ):
+        strategy_key_update_all = ""
+        strategy_key_recreate_all = ""
+        if object == "Models":
+            yml_cfg_destination = destination
+            # If destination does not contain Jinja syntax on last component it's a single file,
+            # and update/recreate must behave as 'update all' or 'recreate all'
+            if not re.search(
+                r"\{\{[A-Za-z]*\}\}", yml_cfg_destination.split("/")[-1].replace(" ", "")
```
Comment (ssassi) [inline dbt_coves/tasks/generate/base.py:187]: This is working only in linux environments. Why you need to set "model_prop_is_single_file" = True?

**Review by ssassi — CHANGES_REQUESTED** [review]

**Review by ssassi — APPROVED** [review]

**Pull request merged**